<?xml version="1.0"?>
<!-- 
  Wrapper launch file for the lebai specific robot_state node.
-->
<launch>
  <!-- IP of robot (or PC running simulation) -->
  <arg name="robot_ip" doc="IP of controller" />
  <arg name="robot_ns" default=""/>
  <arg name="use_ns" default="0"/>
  <!-- <arg name="joint_names_param" default="controller_joint_names"/> -->
  <param name="robot_ip_address" type="str" value="$(arg robot_ip)" unless="$(arg use_ns)"/>
  <param name="$(arg robot_ns)/robot_ip_address" type="str" value="$(arg robot_ip)" if="$(arg use_ns)"/>

  <param name="rcs_port" type="int" value="5181"/>
  <!-- load the correct version of the robot state node -->
  
  <node name="robot_state" pkg="lebai_driver" type="robot_state" unless="$(arg use_ns)" output="screen">
  </node> 

  <node name="$(arg robot_ns)_state" pkg="lebai_driver" type="robot_state" if="$(arg use_ns)" output="screen">
    <remap from="controller_joint_names" to="$(arg robot_ns)/controller_joint_names"/>
    <remap from="feedback_states" to="$(arg robot_ns)/feedback_states"/>
    <remap from="gripper_status" to="$(arg robot_ns)/gripper_status"/>
    <remap from="io_status" to="$(arg robot_ns)/io_status"/>
    <remap from="joint_states" to="$(arg robot_ns)/joint_states"/>
    <remap from="robot_status" to="$(arg robot_ns)/robot_status"/>
    <remap from="robot_ip_address" to="$(arg robot_ns)/robot_ip_address"/>
  </node>    
</launch>